From 6fe3bf045c696adbf7ded71c5b08484f53b8d74b Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Tue, 8 May 2012 17:15:21 +0800
Subject: [PATCH 025/158] powerpc/44x: allow the kernel to be run from a
 non-zero physical address

CONFIG_PHYSICAL_START always be used to set the LOAD program header
physical address field in the resulting ELF image. And the kernel
also uses this address to set the load and entry address when creating
an uImage. So we need to set this physical address properly when we
have to load the uImage to a non-zero physical address.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index fe404e7..2c18425 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -920,7 +920,7 @@ config KERNEL_START
 
 config PHYSICAL_START_BOOL
 	bool "Set physical address where the kernel is loaded"
-	depends on ADVANCED_OPTIONS && FLATMEM && FSL_BOOKE
+	depends on ADVANCED_OPTIONS && FLATMEM && (FSL_BOOKE || 44x)
 	help
 	  This gives the physical address where the kernel is loaded.
 
-- 
1.8.1.2

